dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.ChatCompletion.cs

```diff
@@ -1234,6 +1234,22 @@ private ToolCallingConfig GetToolCallingConfiguration(Kernel? kernel, OpenAIProm
             return new(tools, toolChoice, autoInvoke, maximumAutoInvokeAttempts);
         }
 
+        if (config.Choice == FunctionChoice.Required)
```

Comment:
Same logic as in FunctionChoice.Auto, should this be an `OR` as well, except for the toolChoice setting?

```suggestion
        if (config.Choice == FunctionChoice.Required)
```

Reply:
I plan to do it in a separate "improvements" PR later.